feat: add support for PHP 8

owenvoke opened the pull request:
<!--
Fill in the relevant information below to help triage your issue.

Pick the target branch based on the following criteria:
  * Documentation improvement: master branch
  * Bugfix: master branch
  * QA improvement (additional tests, CS fixes, etc.) that does not change code
    behavior: master branch
  * New feature, or refactor of existing code: develop branch

You MUST provide a signoff in your commits for us to be able to accept your
patch; you can do this by providing either the --signoff or -s flag when using
"git commit". Please see the project contributing guide and
https://developercertificate.org for details.
-->

|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | no
| BC Break      | no
| New Feature   | yes
| RFC           | no
| QA            | no

### Description

This adds support for PHP 8. It looks like Travis doesn't seem to support PHP 8 yet though (I haven't used Travis for ages)?

Closes #3

owenvoke committed:
feat: update to support PHP 8

Signed-off-by: Owen Voke <development@voke.dev>

owenvoke committed:
feat: drop support for PHP <7.3

Signed-off-by: Owen Voke <development@voke.dev>

owenvoke committed:
feat: update to PHPUnit 9.3

Signed-off-by: Owen Voke <development@voke.dev>

owenvoke committed:
ci: update to support PHP 8 in Travis CI

Signed-off-by: Owen Voke <development@voke.dev>

rieschl commented:
> This adds support for PHP 8. It looks like Travis doesn't seem to support PHP 8 yet though (I haven't used Travis for ages)?


Yes you still have to use `nightly`.

owenvoke committed:
ci: fix PHP 8 by using nightly

Signed-off-by: Owen Voke <development@voke.dev>

owenvoke commented:
Thanks, I've now updated it to use Nightly.

froschdesign commented:
@owenvoke 
Thanks for your contribution! 👍 
Please look at the related task list for support PHP 8: #3

owenvoke commented:
I'm following that task list. 👍🏻  Currently on the last one, but it looks like the only issue in the tests is the `servicemanager` which needs to be updated to a minimum of `3.2.1` as far as I can tell. Is that ok to do? Or do we still need to support `2.x`, which doesn't seem to have the changes.

froschdesign commented:
> I'm following that task list.

You have missed an important task:

> Modify `composer.json` to drop support for PHP less than 7.3

Which also means: remove the old version from the Travis configuration. 😉

owenvoke committed:
chore: remove old versions of Laminas dependencies

Signed-off-by: Owen Voke <development@voke.dev>

rieschl commented on composer.json:
```diff
@@ -25,15 +25,15 @@
     },
     "require": {
-        "php": "^5.6 || ^7.0",
+        "php": "^7.3 || ^8.0",
```
Please use `~8.0.0` as mentioned in the issue. That way the package supports PHP up to `8.0.*`.

owenvoke commented:
Ah, makes sense. 🤦🏻 I've removed the old versions, and updated to a minum of Service Manager v3.4, which added PHP8 support (and dropped <7.3). And also now removed the old versions from Travis. 😬 Apologies for forgetting this.

owenvoke committed:
chore: update to use tilde for PHP 8 version

Signed-off-by: Owen Voke <development@voke.dev>

owenvoke committed:
ci: remove PHP lower than 7.3 from Travis

Signed-off-by: Owen Voke <development@voke.dev>

owenvoke committed:
ci: add back missing test coverage and cs check

Signed-off-by: Owen Voke <development@voke.dev>

boesing committed:
Add `CHANGELOG` entry

Signed-off-by: Maximilian Bösing <2189546+boesing@users.noreply.github.com>

boesing commented:
Thanks @owenvoke
